iqama_fetcher.py

```diff
@@ -1,42 +1,114 @@
-"""Fetches iqama times from our heroku server."""
+"""Fetches iqamah times from the Iqamah.net service."""
 
-import json
+import re
+from xml.etree import ElementTree
 import util
 import requests
+from masjid_util import GetIqamaID
 
-_SCRAPER_URL = 'https://islam-buddy-staging.herokuapp.com/iqama'
-
+_IQAMAH_NET_URL = 'http://feed.iqamah.net/IQ'
 
 def GetIqamaTime(desired_prayer, masjid):
-  """Gets the iqama time from the heruko scraper service by peforming a POST.
+  """Gets the iqama time from the iqamah.net service by peforming a GET.
 
   Args:
     desired_prayer: DailyPrayer enum representing desired salah
     masjid: a string representing the masjid of intereset
 
-  Returns: a dict containing the prayers and iqama times
+  Returns: a string containing the iqama time along with AM/PM
   """
 
-  # set up the parameters in the format expected by the scraper service
-  post_params = {
-      'masjid': masjid,
-  }
-  print 'scraper get_params = ', post_params
-  request = requests.post(_SCRAPER_URL, params=post_params, timeout=15)
+  # making sure the masjid input is in UTF-8 format
+  masjid_encoded = util.EncodeParameter(masjid, 0)
+  # getting the id from the _MASJID_METADATA
+  masjid_id = GetIqamaID(masjid_encoded)
```

Comment:
you can just do GetIqamaId(util.EncodeParam(masjid))

also i don't really like the 0 there - i think that should be a default